### ISSUE TITLE
Cannot input text into  formfield

### ISSUE THREAD

**Issue by nirtuj** [OnionBrowser/iOS-OnionBrowser#137]
This happens on https://mail.riseup.net/sm/src/webmail.php, a webmail service using squirrelmail.
When clicking reply on a message I can put the focus onto the text field, yet it is impossible to enter any text (the on screen keyboard is shown, yet no keypresses get into to the textfield). This works in mobile safari.

**Comment by tladesignz** [comment]
Ok, so I got ahold of a Riseup invite, which was less annoying to get than actually install this SquirrelMail desaster. This thing is pretty old and Riseup currently uses RoundCube as default and only has SquirrelMail as a fallback option.

It's a really old application with no explicit JavaScript resources or any XHR loading or similar. Frames, some CSS and very limited inline JavaScript.

What the reporter actually meant, was the `textarea` field containing the mail body and I can confirm strange behavior.

I had a large "welcome message" to test with. What I noticed, is, that I could always delete stuff and only randomly add characters. I can add characters when I set focus outside of the textarea and back to it. As long as I don't reposition the cursor, I can add text. After reposition, I can only delete.

When composing a new mail, that problem doesn't exist.

When using RoundCube, that problem doesn't exist.

I found an interesting snippet in SquirrelMail's JavaScript:

```JavaScript
var alreadyFocused = false;
function checkForm() {

    if (alreadyFocused) return;
    var f = document.forms.length;
    var i = 0;
    var pos = -1;
    while( pos == -1 && i < f ) {
        var e = document.forms[i].elements.length;
        var j = 0;
        while( pos == -1 && j < e ) {
            if ( document.forms[i].elements[j].type == 'text' ) {
                pos = j;
            }
            j++;
        }
        i++;
    }
    if( pos >= 0 ) {
        document.forms[i-1].elements[pos].focus();
    }
}
```

Along with a lot of 

```html
 onfocus="alreadyFocused=true;" 
```

This may well be the root cause of the issue, where `onfocus` events of HTML elements don't interact well with the code inside `UIWebView`.

The difference to mobile Safari (which I can also confirm) could be explained due to different engine. (Newer `WKWebView` vs. our older `UIWebView`.)

@mtigas Or do you think, we have some scripts injected, which interfere? I can hardly believe that, but you know that better than me.

**Comment by mtigas** [comment]
Interesting. I tested it and it worked, but I also only tested using short message strings.

Probably worth testing if it works in vanilla Endless and/or another browser using UIWebView? Just to figure out whether it's specific to us or not. (@tladesignz I'll probably do that tonight (US Eastern time) if you haven't gotten to it yet.)

**Comment by tladesignz** [comment]
Please do, I'm done for today.

**Comment by tladesignz** [comment]
Ugh. Latest vanilla Endless works fine.
I suggest to first do #128, and then revisit this.

**Comment by tladesignz** [comment]
Can confirm: works with latest updates from Endless upstream. So, this will be resolved with #128.

**Comment by tladesignz** [comment]
Fixed by fixing #128.
